core/services/relay/evm/mercury_provider.go

```diff
@@ -103,3 +108,37 @@ func (p *mercuryProvider) ContractTransmitter() ocrtypes.ContractTransmitter {
 func (p *mercuryProvider) MercuryServerFetcher() relaymercury.MercuryServerFetcher {
 	return p.transmitter
 }
+
+func (p *mercuryProvider) ChainReader() relaymercury.ChainReader {
+	return p.chainReader
+}
+
+var _ relaymercury.ChainReader = (*chainReader)(nil)
+
+type chainReader struct {
+	tracker httypes.HeadTracker
+}
+
+func NewChainReader(h httypes.HeadTracker) relaymercury.ChainReader {
+	return &chainReader{
+		tracker: h,
+	}
+}
+
+func (r *chainReader) LatestBlocks(ctx context.Context, k int) ([]relaymercury.Block, error) {
+	evmBlocks := r.tracker.LatestChain().AsSlice(k)
+	if len(evmBlocks) == 0 {
+		return nil, fmt.Errorf("no blocks available")
```

Comment:
Hmm explicit error here seems like it would make things difficult for the caller. Why don't we leave it up to them to decided if this is an error or not?
```suggestion
		return nil, nil
```

Reply:
Or we could generalize to checking length against `k`, and then reconsider this condition in that context.

Reply:
Agreed. thanks!